api docs: Improve how arguments are handled.

YagoGG opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing Plan:** lint and mypy checks passed locally. Also, in order to make sure that the changes worked properly, I removed the following from `zerver/openapi/zulip.yaml`:

~~~diff
diff --git a/zerver/openapi/zulip.yaml b/zerver/openapi/zulip.yaml
index 2aa379797..dadb6ef5e 100644
--- a/zerver/openapi/zulip.yaml
+++ b/zerver/openapi/zulip.yaml
@@ -98,14 +98,6 @@ paths:
   /users/me/{stream_id}/topics:
     get:
       description: Get all the topics in a specific stream.
-      parameters:
-      - name: stream_id
-        in: path
-        description: The unique ID of the stream.
-        schema:
-          type: integer
-        example: 42
-        required: true
       security:
       - basicAuth: []
       responses:
~~~

This way, I could compare between these two:

- http://localhost:9991/api/add-subscriptions (which has arguments)
- http://localhost:9991/api/get-stream-topics (which now should have no arguments and thus display the message, only for testing purposes).

**GIFs or Screenshots:**

![Endpoint with no args](https://user-images.githubusercontent.com/7356565/42633441-23dbed18-85fe-11e8-8623-0bea90aaaf5c.png)

YagoGG commented:
BTW, after noticing that `text` isn't a string but a list here:

~~~diff
+                    if arguments:
+                        text = self.render_table(arguments)
+                    else:
+                        text = ['This endpoint does not consume any '
+                                'arguments.']
~~~

I wonder whether it may be worth making `render_table()` return a string, by working with string concatenation (rather than list's `.append()`). Would love to know your thoughts on this, @eeshangarg!

eeshangarg commented on zerver/lib/bugdown/api_arguments_table_generator.py:
```diff
@@ -48,26 +48,33 @@ def run(self, lines: List[str]) -> List[str]:
+                            if e.args != ('parameters',):
+                                raise e
+
+                            arguments = []
```
This is not a huge deal but could you please move this declaration to before the `try` statement? I think it is less confusing. :)

YagoGG commented:
Thanks a lot for reviewing, @eeshangarg! I just updated the PR addressing your inline comment (and also added one small commit fixing something arguments-related).

About changing the docs of endpoints that accept no arguments: if you notice I only put the `try/catch` statement around the OpenAPI part. That's because we're about to finish the migration to OpenAPI —it's already done, we only have to merge it—, and adding the extra logic for something that we are about to change and only affects two endpoints didn't seem worthwhile. Let me know if you think otherwise :)

eeshangarg commented:
@YagoGG: Cool, that makes sense! :) This LGTM! 

@timabbott: Could you please merge this? Thanks! :)

timabbott commented:
Merged as the series of commits ending in 40efac2193b17ddfe5dbce1b3089f771e8e7c5ff, after adding a commit to use early return to better implement the line-wrapping you had in the second commit.  Thanks folks!
